ui/src/app/views/workspace-edit/component.html

```diff
@@ -106,10 +109,21 @@ <h4>Request a Review of Your Research Purpose</h4>
     </div>
     <div style="display: flex; flex: 0 0 0; padding: 0">
       <div class="button">
-        <button *ngIf="adding else editing" class="btn add-button btn-primary" [clrLoading]="savingWorkspace" (click)="addWorkspace()">Create Workspace</button>
-        <ng-template #editing>
-          <button class="btn add-button btn-primary" [clrLoading]="savingWorkspace" (click)="updateWorkspace()" [disabled]="!hasPermission">Update Workspace</button>
-        </ng-template>
+        <ng-container [ngSwitch]="mode">
+          <button *ngSwitchCase="Mode.Create" class="btn add-button btn-primary"
+                  [clrLoading]="savingWorkspace" (click)="addWorkspace()">
```

Comment:
Not really important, but worth mentioning: I usually prefer an indent (4 spaces, say) over this kind of alignment because that pattern has more graceful degradation in the face of `really-super-long-tag-name` and likely has slightly wider support in text editors.

Reply:
Yep, done, good call. Not sure how I wound up with 8 spaces here.